GVRf/Framework/jni/objects/textures/cubemap_texture.h

```diff
@@ -33,35 +34,11 @@ class CubemapTexture: public Texture {
     explicit CubemapTexture(JNIEnv* env, jobjectArray bitmapArray,
             int* texture_parameters) :
             Texture(new GLTexture(TARGET, texture_parameters)) {
-        glBindTexture(TARGET, gl_texture_->id());
-        for (int i = 0; i < 6; i++) {
-            jobject bitmap = env->GetObjectArrayElement(bitmapArray, i);
-
-            AndroidBitmapInfo info;
-            void *pixels;
-            int ret;
-
-            if (bitmap == NULL) {
-                std::string error =
-                        "new BaseTexture() failed! Input bitmap is NULL.";
-                throw error;
-            }
-            if ((ret = AndroidBitmap_getInfo(env, bitmap, &info)) < 0) {
-                std::string error = "AndroidBitmap_getInfo () failed! error = "
-                        + ret;
-                throw error;
-            }
-            if ((ret = AndroidBitmap_lockPixels(env, bitmap, &pixels)) < 0) {
-                std::string error =
-                        "AndroidBitmap_lockPixels () failed! error = " + ret;
-                throw error;
-            }
+        pending_gl_task_ = GL_TASK_INIT_BITMAP;
+        env_ = env;
 
-            glTexImage2D(GL_TEXTURE_CUBE_MAP_POSITIVE_X + i, 0, GL_RGBA,
-                    info.width, info.height, 0, GL_RGBA, GL_UNSIGNED_BYTE,
-                    pixels);
-
-            AndroidBitmap_unlockPixels(env, bitmap);
+        for (int i = 0; i < 6; i++) {
```

Comment:
Do you think that this class needs a destructor that releases the references just in case - if for whatever reasons runPendingGL doesn't get to run before destruction? It probably can't happen in practice now but theoretically it might.